cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/integration_utils_test.go

```diff
@@ -0,0 +1,337 @@
+					Name:      name,
+					Namespace: namespace,
+				},
+			},
+			RoleRef: kuberbacv1.RoleRef{
+				Kind: "ClusterRole",
+				Name: "cluster-admin",
+			},
+		},
+		metav1.CreateOptions{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func kubeDeleteServiceAccount(t *testing.T, name, namespace string) error {
+	t.Logf("+kubeDeleteServiceAccount(%s,%s)", name, namespace)
+	if typedClient, err := kubeGetTypedClient(); err != nil {
+		return err
+	} else if err = typedClient.RbacV1().ClusterRoleBindings().Delete(
+		context.TODO(),
+		name+"-binding",
+		metav1.DeleteOptions{}); err != nil {
+		return err
+	} else if err = typedClient.CoreV1().ServiceAccounts(namespace).Delete(
+		context.TODO(),
+		name,
+		metav1.DeleteOptions{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func kubeDeleteNamespace(t *testing.T, namespace string) error {
+	t.Logf("+kubeDeleteNamespace(%s)", namespace)
+	if typedClient, err := kubeGetTypedClient(); err != nil {
+		return err
+	} else if typedClient.CoreV1().Namespaces().Delete(
+		context.TODO(),
+		namespace,
+		metav1.DeleteOptions{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func kubeGetHelmReleaseResourceInterface(namespace string) (dynamic.ResourceInterface, error) {
+	clientset, err := kubeGetDynamicClient()
+	if err != nil {
+		return nil, err
+	}
+	relResource := schema.GroupVersionResource{
+		Group:    fluxHelmReleaseGroup,
+		Version:  fluxHelmReleaseVersion,
+		Resource: fluxHelmReleases,
+	}
+	return clientset.Resource(relResource).Namespace(namespace), nil
+}
+
+func kubeGetHelmRepositoryResourceInterface(namespace string) (dynamic.ResourceInterface, error) {
+	clientset, err := kubeGetDynamicClient()
+	if err != nil {
+		return nil, err
+	}
+	repoResource := schema.GroupVersionResource{
+		Group:    fluxGroup,
+		Version:  fluxVersion,
+		Resource: fluxHelmRepositories,
+	}
+	return clientset.Resource(repoResource).Namespace(namespace), nil
+}
+
+func kubeGetDynamicClient() (dynamic.Interface, error) {
+	if dynamicClient != nil {
+		return dynamicClient, nil
+	} else {
+		if config, err := restConfig(); err != nil {
+			return nil, err
+		} else {
+			dynamicClient, err = dynamic.NewForConfig(config)
+			return dynamicClient, err
+		}
+	}
+}
+
+func kubeGetTypedClient() (kubernetes.Interface, error) {
+	if typedClient != nil {
+		return typedClient, nil
+	} else {
+		if config, err := restConfig(); err != nil {
+			return nil, err
+		} else {
+			typedClient, err = kubernetes.NewForConfig(config)
+			return typedClient, err
+		}
+	}
+}
+
+func restConfig() (*rest.Config, error) {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	return clientcmd.BuildConfigFromFlags("", kubeconfig)
+}
+
+func randSeq(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+// global vars
+var letters = []rune("abcdefghijklmnopqrstuvwxyz0123456789")
+var typedClient kubernetes.Interface
+var dynamicClient dynamic.Interface
```

Comment:
Do we need global vars? Normally it should be possible to avoid this type of state (your kubeGet* fns above already return the clients).

Reply:
not sure I get what you're asking. What's the alternative to global vars if I want to keep some state in between calls (other than to have to declare a new receiver struct or explicitly pass in the state back and forth, yuck)? Must be some language feature I am not aware of. Please tell me.

Reply:
I'm not asking you to change anything here, but worth reading: https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables

Reply:
Sure, I'll give a read. Just so we are on the same page, this is test code we are talking about, not production. Furthermore, this is test code that you've pointed out will become obsolete at some point not too far away. So going an extra mile (or in this case an extra 10 miles based on what I read of the article so far) and taking a shortcut seemed like an acceptable thing to do for that case. If you find any cases where I am doing it in production code, please point them out

Reply:
Yep, as above, nothing to change here. Just worth being aware, that's all. I like the last line ("the bottom line") of Dave's post.